### ISSUE TITLE
Error: " The message port closed before a response was received "

### ISSUE THREAD

**Issue by charmful0x** [th8ta/ArConnect#9]
Accessing permissions works fine, but an error get thrown:
`Unchecked runtime.lastError: The message port closed before a response was received`

And therefore, this error cause error in transaction's signing : `Uncaught (in promise) Error signing transaction`

I thought it may be due to end-user syntax error, however it seems a mishandled async response to `runtime.sendMessage` as i checked another dapp using ArConnect as the main and only login option (found on Arweave discord: https://arweave.net/AEn40YjdI2FEeK66pZfBljkHYgmGTD8ONWNhIqbAk3U ) and it has similar logged errors.

**Comment by martonlederer** [comment]
This error is caused by the background script's message sending. It does not affect the signing functionalities, so something else is wrong with your transaction signing.

I tried fixing this error, but we don't really know why it keeps happening, just ignore it

**Comment by charmful0x** [comment]
i have even tried a very simple tx creation for testing, same issue at signing

```
 async function doAction() {
 
    let transaction = await arweave.createTransaction( {data: 'hello'} );

    transaction.addTag("Content-Type", "text/plain");

    await arweave.transactions.sign(transaction);

    await arweave.transactions.post(transaction)

    console.log(transaction.id)
}
```

the `"SIGN_TRANSACTION"` permission is granted

**Comment by martonlederer** [comment]
As mentioned above, this message doesn't affect the signing. You must have something else that's not working in your code

**Comment by martonlederer** [comment]
@charmful0x let me know if you have managed to figure out your problem :)

**Comment by martonlederer** [comment]
Hey hey, the port error has been fixed in `0.2.5`. Closing this, feel free to open a new issue if you are still having problems

**Comment by charmful0x** [comment]
Thank you!
